flang/include/flang/Tools/CLOptions.inc

```diff
@@ -153,12 +153,12 @@ inline void createDefaultFIROptimizerPassPipeline(mlir::PassManager &pm) {
   // simplify the IR
   mlir::GreedyRewriteConfig config;
   config.enableRegionSimplification = false;
-  fir::addCSE(pm);
+  pm.addPass(mlir::createCSEPass());
```

Comment:
I'm kind of thinking, since these are not really the same, that getting rid of FIR's CSE at this time is premature in the fir-dev branch.

It's fine upstream, but I don't think we want to reduce performance or regress the working branch, for example.

Reply:
At some point we have to make sure upstream and fir-dev are doing the same things. I'm ok to wait to make the change. Upstream is not using any CSE so far. I have a patch ready to introduce it.

Reply:
> Much of this is fine, but I'm not sure ripping out FIR's CSE in the fir-dev branch and replacing it with a weaker pass is needed.

Fine for me to wait. After D123492 we will just miss the `pure` and `volatile` attributes. I'm not sure how much we rely on this today in fir-dev. I'm working on a patch to have this upstream with the MLIR CSE so if it is needed we can wait until it's in.